Scripts/Tracking/Collision/CollisionList.cs

```diff
@@ -0,0 +1,89 @@
+﻿namespace VRTK.Core.Tracking.Collision
+{
+    using UnityEngine;
+    using UnityEngine.Events;
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Holds a collection of collisions created by a <see cref="CollisionNotifier"/>.
+    /// </summary>
+    public class CollisionList : MonoBehaviour
+    {
+        /// <summary>
+        /// Defines the event with the <see cref="List{CollisionNotifier.EventData}"/>.
+        /// </summary>
+        [Serializable]
+        public class UnityEvent : UnityEvent<List<CollisionNotifier.EventData>>
+        {
+        }
+
+        /// <summary>
+        /// Emitted when the collision collection is no longer empty.
+        /// </summary>
+        public UnityEvent NotEmpty = new UnityEvent();
+        /// <summary>
+        /// Emitted when the collision collection contents has changed.
+        /// </summary>
+        public UnityEvent Changed = new UnityEvent();
+        /// <summary>
+        /// Emitted when the collision collection has become empty.
+        /// </summary>
+        public UnityEvent Empty = new UnityEvent();
+
+        /// <summary>
+        /// The collisions currently stored in the collection.
+        /// </summary>
+        public List<CollisionNotifier.EventData> Elements
+        {
+            get;
+            protected set;
+        } = new List<CollisionNotifier.EventData>();
+
+        /// <summary>
+        /// Determines if the collection is empty.
+        /// </summary>
+        /// <returns><see langword="true"/> if the collection is empty.</returns>
+        public virtual bool IsEmpty()
+        {
+            return (Elements.Count == 0);

```

Comment:
Do we really need to add a method for this? I'd rather have everyone type this out than introduce another method which allows to be overridden and thus break any usage of the list as the count and this empty check can get out of sync.

tl;dr: I don't see this saving us a lot so I wonder if it's worth removing it for the sake of ensuring the usage of the list is sound.